db/sqlinterfaces.c

```diff
@@ -1233,6 +1233,8 @@ static void sql_statement_done(struct sql_thread *thd, struct reqlogger *logger,
 
     thd->crtshard = 0;
 
+    reqlog_set_api_type(logger, clnt->plugin.api_type(clnt));
```

Comment:
Should we do this earlier (`run_stmt` perhaps?) We may log a query even before it has finished.

Reply:
Try running `select sleep(10)`. It should log twice. Once after the 5s threshold. And once on completion.

Reply:
It looks like the logger that is used to print stats about a running request is different from the one that prints stats about a completed request. `run_stmt` has access to the logger that is used to print stats about a completed request. `reqlog_long_running_clnt` is the function that sets up the logger when stats get printed during a long request, so I set the api type [there](https://github.com/bloomberg/comdb2/pull/4130/files#diff-6b1a6f166109d69c771d205177e9f5c0769e9b6cdc41fb93e54f45393c6d3d76R1946). 

I also moved this line to [`query_stats_setup`](https://github.com/bloomberg/comdb2/pull/4130/files#diff-c2934cd4869e4f80bc2ffe6bcb07cca56937370c1f6fbcfce0fe3d17388423feR2782) because it makes more sense to set the client api during initial setup. I don't think that this changes any behavior though since this value doesn't appear to be printed until the request completes.